os/kernel/pthread/pthread_keycreate.c

```diff
@@ -125,30 +125,40 @@
  *
  ****************************************************************************/
 
-int pthread_key_create(FAR pthread_key_t *key, CODE void (*destructor)(void *))
+int pthread_key_create(FAR pthread_key_t *key, pthread_destructor_t destructor)
 {
 #if CONFIG_NPTHREAD_KEYS > 0
-	FAR struct tcb_s *rtcb = this_task();
-	FAR struct task_group_s *group = rtcb->group;
+	struct pthread_tcb_s *rtcb = (struct pthread_tcb_s *)this_task();
+	struct task_group_s *group = rtcb->cmn.group;
+
+	int key_index = 0;
 	int ret = EAGAIN;
 
 	DEBUGASSERT(group);
 
 	/* Check if we have exceeded the system-defined number of keys. */
 
-	if (group->tg_nkeys < PTHREAD_KEYS_MAX) {
-		/* Return the key value */
+	while (key_index < PTHREAD_KEYS_MAX) {
+		/* Find not in-use key */
+
+		if (!group->tg_keys[key_index]) {
```

Comment:
is it okay when key_index is smaller than PTHREAD_KEYS_MAX and greater than CONFIG_NPTHREAD_KEYS?